build.ps1

```diff
@@ -1022,6 +1026,126 @@ function Ensure-Certificates {
     }
 }
 
+function Ensure-Crypto-File {
+    param(
+        [string]$conf_dir
+    )
+
+    $DEPLOYMENT_FILE = Join-Path $conf_dir "deployment.yaml"
+    # Resolve the .. path segment to get a clean key directory path
+    $KEY_DIR_Temp = Join-Path $conf_dir ".." "resources/security"
+    $KEY_DIR = (Resolve-Path -Path $KEY_DIR_Temp).Path
+    $KEY_FILE = Join-Path $KEY_DIR "crypto.key"
+    $KEY_PATH_IN_YAML = "repository/resources/security/crypto.key"
+
+    Write-Host "================================================================"
+    Write-Host "Ensuring crypto key file exists..."
+
+    # 1. Check if the key file exists
+    if (Test-Path $KEY_FILE) {
+        Write-Host "Default crypto key file already present in $KEY_FILE. Skipping generation."
+    }
+    else {
+        Write-Host "Default crypto key file not found. Generating new key at $KEY_FILE..."
+        $NEW_KEY = $null
+        
+        # --- START: OpenSSL-first logic ---
+        # Try OpenSSL first, just like Ensure-Certificates
+        $openssl = Get-Command openssl -ErrorAction SilentlyContinue
+        if ($openssl) {
+            try {
+                Write-Host " - Using OpenSSL to generate key..."
+                # openssl rand -hex 32 returns a 64-char string.
+                # Use Out-String | Trim to capture it cleanly.
+                $NEW_KEY = (openssl rand -hex 32 | Out-String).Trim()
+                
+                if ($LASTEXITCODE -ne 0 -or [string]::IsNullOrEmpty($NEW_KEY)) {
+                    throw "OpenSSL rand command failed or returned empty."
+                }
+            }
+            catch {
+                Write-Host " - OpenSSL failed: $_. Falling back to .NET cryptography."
+                $NEW_KEY = $null
```

Comment:
Create the key using posix tools if openssl not found. Or else we could create the key without using openssl altogether.